content/en/docs/ops/configuration/traffic-management/network-topologies/index.md

```diff
@@ -0,0 +1,186 @@
+weight: 60
+keywords: [traffic-management,ingress,gateway]
+---
+
+## (Experimental) Configuring network topologies
+
+Istio provides the ability to manage settings like X-Forwarded-For (XFF) and X-Forwarded-Client-Cert (XFCC), which are
+dependent on how the gateway workloads are deployed. This is currently an experimental feature.
+
+Many users choose to deploy Istio ingress gateways in using various network topologies
+(e.g. behind Cloud Load Balancers, a self-managed Load Balancer or directly expose the
+Istio ingress gateway to the internet). As such, these topologies require different ingress gateway configurations for
+transporting correct client attributes like IP/certs to the workloads running in the cluster.
+
+Configuration of XFF and XFCC headers can be configured by using `MeshConfig` during Istio
+*installation* or by adding a pod annotation.
+
+To simplify configuring network topology during installation create a single YAML file to pass to `istioctl`:
+
+{{< text yaml >}}
+cat <<'EOF' > topology.yaml
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      gatewayTopology:
+EOF
+{{< /text >}}
+
+{{< warning >}}
+If Istio ingress gateway was already running prior to application of the `MeshConfig` you will need
+to restart any Istio ingress gateway pods.
+{{< /warning >}}
+
+Both of the settings discussed can also be configured using the `proxy.istio.io/config` annotation to the Pod spec
+of your Istio ingress gateway.
+
+{{< text yaml >}}
+...
+  metadata:
+    annotations:
+      "proxy.istio.io/config": '{"gatewayTopology" : { "numTrustedProxies": 2 } }'
+{{< /text >}}
+
+### Configuring X-Forwarded-For Headers
+
+Many applications, both old and new, rely on client attributes, such as X-Forward-For header,
+to be forwarded by reverse proxies in a request. However, due to the variety of network
+topologies Istio can be deployed in, it is required to set the number of trusted proxies deployed in front
+of the Istio gateway proxy so that the client address can be extracted correctly.
+
+To set the number of trusted proxies add the following to your `topology.yaml` file.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      gatewayTopology:
+        numTrustedProxies: <VALUE>
+{{< /text >}}
+
+For example, if you have a cloud based LB, a reverse proxy and then the Istio gateway proxy
+then `<VALUE>` would be 2.
+
+{{< warning >}}
+Note that all the proxies in front need of Istio gateway proxy must parse HTTP traffic and append the X-Forwarded-For
+headers at each hop. If the number of entries in X-Forwarded-For header is less than the number of
+trusted hops configured, Envoy falls back to using the immediate downstream address as the trusted
+client address. Please refer to [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for)
```

Comment:
this x-forward-for link should be right at the front of the usage of x-forward-for

Reply:
Done